### DISCUSSION TITLE
package published to pypi is missing project description.

### DISCUSSION THREAD

**Discussion by BeyondEvil** [python-poetry/poetry#5345] [Q&A] [answered]
My previous release shows the long description in pypi.

My latest release, using `poetry publish`, does not.

How can I fix this?

https://github.com/pytest-dev/pytest-metadata/blob/master/.github/workflows/publish.yml#L29-L34

https://github.com/pytest-dev/pytest-metadata/blob/master/setup.py

https://github.com/pytest-dev/pytest-metadata/blob/master/setup.cfg#L4-L5

**Comment by abn** [comment, marked as answer]
You should probably add `readme = "README.rst"` to your `pyproject.toml` file.

Additionally, best to drop both the `setup.py` and `setup.cfg` files as they are only required for `setuptools`. Move the metadata to the `tool.poetry` section.

You should probably also update your `build-system` section to be something like https://github.com/python-poetry/poetry/blob/15f6ec1758d783d4cb3070a8027725bc611c0a90/pyproject.toml#L74-L76

You can test all this by locally doing `poetry build` then checking metadata in the wheel generated.

**Reply by BeyondEvil** [reply]
Thanks,

I've tried desperately to understand what the downsides/side effects are to dropping the setup.* way of doing things. Is there a tl;dr?

**Reply by abn** [reply]
Depends on your project really. The answer that applies to most projects out there is that there are no downsides per say unless you use a particular feature of setuptools that poetry does not implement (eg: I am not sure how `tool.setuptools_scm` interacts with a poetry project). The `setup.*` files are for when you use setuptools for your project build and packaging. 

Maybe also see https://packaging.python.org/en/latest/guides/tool-recommendations/

**Reply by BeyondEvil** [reply]
Is there somewhere I can read and understand this, preferably dumbed down - I understand nothing when reading PEPs. 🤷‍♂️ 

Looking at Poetry it seems like there's not a 1-to-1 between setup.cfg and the poetry section in pyproject.toml.

**Reply by abn** [reply]
I'd recommend the Poetry docs as as starting point - https://python-poetry.org/docs/. Hopefully that is less PEPy. Probably best to look at what you need and configure that rather than try copy 1-1.

Looking at google there are a few posts like this that might help - https://browniebroke.com/blog/migrating-project-to-poetry/.

**Reply by BeyondEvil** [reply]
Reading, thanks.

I've gotten two issues reported against my project since moving to Poetry. I'm trying to understand them.

In https://github.com/pytest-dev/pytest-metadata/pull/41, why doesn't Poetry successfully set the entrypoint - since it's defined correctly in `setup.cfg`.

And why did tests stop being included in the sdist tarball, https://github.com/pytest-dev/pytest-metadata/issues/42 ?

Thanks! 🙏

**Reply by abn** [reply]
The `setup.cfg` is useless for a poetry project and it is never read at all.

The fix is to define entrypoint plugins as documented at https://python-poetry.org/docs/pyproject/#plugins

As for including the tests, us the include statement. 

https://github.com/python-poetry/poetry-core/blob/213e3670b6287053b350e8c29972e0ea2e7526a5/pyproject.toml#L24-L26

**Reply by BeyondEvil** [reply]
> The `setup.cfg` is useless for a poetry project and it is never read at all.
> 
Even tho I define this under the build? https://github.com/pytest-dev/pytest-metadata/blob/master/pyproject.toml#L1-L8

**Reply by abn** [reply]
If you use `poetry build` the `build-system` section will be ignored. It's only used when a PEP517 frontend like pip or build is used. And if you have setuptools in there, poetry-core won't be used to build the project, but setuptools. That causes all sorts of other problems and inconsistencies. TL;DR is it needs to be as I indicated in my first comment.

**Reply by BeyondEvil** [reply]
> If you use `poetry build` the `build-system` section will be ignored. It's only used when a PEP517 frontend like pip or build is used. And if you have setuptools in there, poetry-core won't be used to build the project, but setuptools. That causes all sorts of other problems and inconsistencies. TL;DR is it needs to be as I indicated in my first comment.

Ok, it's starting to make sense now. 😊 

Thanks for taking the time to explain it. 🙇

**Reply by abn** [reply]
@BeyondEvil added https://github.com/pytest-dev/pytest-metadata/pull/44 to make things a bit clearer. (hopefully)
